discovery-provider/src/queries/get_trending_tracks.py

```diff
@@ -1,18 +1,40 @@
 import logging  # pylint: disable=C0302
 import datetime
 from dateutil.parser import parse
-from flask.globals import request
 
 from src.utils.db_session import get_db_read_replica
 from src.queries.get_unpopulated_tracks import get_unpopulated_tracks
 from src.queries.query_helpers import populate_track_metadata, \
     get_users_ids, get_users_by_id
 from src.tasks.generate_trending import generate_trending
-from src.utils.redis_cache import extract_key, use_redis_cache
+from src.utils.redis_cache import use_redis_cache
 
 logger = logging.getLogger(__name__)
 
-SCORES_CACHE_DURATION_SEC = 10 * 60
+TRENDING_LIMIT = 100
+
+def make_trending_cache_key(time_range, genre):
+    """Makes a cache key resembling `generated-trending:week:electronic`"""
+    return f"generated-trending:{time_range}:{(genre.lower() if genre else '')}"
+
+def generate_unpopulated_trending(session, genre, time_range):
+    trending_tracks = generate_trending(
+        session, time_range, genre,
+        TRENDING_LIMIT, 0)
+
+    track_scores = [z(time_range, track) for track in trending_tracks['listen_counts']]
+    sorted_track_scores = sorted(track_scores, key=lambda k: k['score'], reverse=True)
+
+    track_ids = [track['track_id'] for track in sorted_track_scores]
+
+    tracks = get_unpopulated_tracks(session, track_ids)
+    return (tracks, track_ids)
+
+def make_generate_unpopulated_trending(session, genre, time_range):
+    def wrapped():
```

Comment:
jw why this is necessary?

Reply:
so if you look at how we call this, it's called in use_redis_cache, which expects a function with no arguments. `make_generate_unpopulated_trending` returns a wrapped function with no arguments that calls generate with the args we want. I can add a comment explaining